src/reconcile/rec_visibility.c

```diff
@@ -746,18 +770,7 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
         supd_restore = F_ISSET(r, WT_REC_EVICT) &&
           (has_newer_updates || F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
 
-        /*
-         * The total update size only contains uncommitted updates. Add the size for the rest of the
-         * chain.
-         *
-         * FIXME-WT-9182: figure out what should be included in the calculation of the size of the
-         * saved update chains.
-         */
-        if (onpage_upd != NULL) {
-            for (upd = tombstone != NULL ? tombstone : onpage_upd; upd != NULL; upd = upd->next)
-                if (upd->txnid != WT_TXN_ABORTED)
-                    upd_memsize += WT_UPDATE_MEMSIZE(upd);
-        }
+        upd_memsize = __rec_calc_upd_memsize(onpage_upd, tombstone, upd_memsize);
```

Comment:
Instead of creating another function, how about moving this logic inside the `__rec_update_save` function. All the required variables that are required are already passed to that function.

Reply:
Calculate the size of upd memsize seems not the task for __rec_update_save based on it's name. I prefer it to be separate.